components/services/markdown/src/index.js

```diff
@@ -28,6 +28,18 @@ class ServiceMarkdown extends Component {
     )
   }
 
+  componentDidUpdate() {
+    const id = document.location.hash.substring(1)
+    if (id) {
+      const element = document.getElementById(id)
+      element && element.scrollIntoView({block: 'start', behavior: 'smooth'})
+    }
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return nextState.html !== this.state.html
```

Comment:
I think we could simplify this kind of comparison. Instead comparing the huge html, creating a new state and using it in the shouldComponentUpdate to only do the action once.